packages/sanity/src/core/releases/tool/detail/events/getReleaseEditEvents.ts

```diff
@@ -0,0 +1,165 @@
+ * I anticipate this would be a rather small number of transactions, given the release document is "small" and shouldn't change much.
+ *
+ * We need to fetch all of them to create the correct pagination of events in the activity feed, given we need to combine this with the
+ * releaseActivityEvents that will be fetched from the events api.
+ */
+function getReleaseTransactions({
+  documentId,
+  client,
+  toTransaction,
+}: {
+  documentId: string
+  client: SanityClient
+  toTransaction: string
+}): Observable<TransactionLogEventWithEffects[]> {
+  const cacheKey = `${documentId}`
+  const cachedTransactions = documentTransactionsCache[cacheKey] || []
+  if (cachedTransactions.length > 0 && cachedTransactions[0].id === toTransaction) {
+    return of(cachedTransactions)
+  }
+
+  function fetchLogs(options: {
+    fromTransaction?: string
+    toTransaction: string
+  }): Observable<TransactionLogEventWithEffects[]> {
+    return from(
+      getTransactionsLogs(client, documentId, {
+        tag: 'sanity.studio.release.history',
+        effectFormat: 'mendoza',
+        limit: TRANSLOG_ENTRY_LIMIT,
+        reverse: true,
+        fromTransaction: options.fromTransaction,
+        toTransaction: options.toTransaction,
+      }),
+    )
+  }
+
+  return fetchLogs({fromTransaction: cachedTransactions[0]?.id, toTransaction: toTransaction})
+    .pipe(
+      expand((response) => {
+        // Fetch more if the transactions length is equal to the limit
+        if (response.length === TRANSLOG_ENTRY_LIMIT) {
+          // Continue fetching if nextCursor exists, we use the last transaction received as the cursor.
+          return fetchLogs({
+            fromTransaction: undefined,
+            toTransaction: response[response.length - 1].id,
+          })
+        }
+        // End recursion by emitting an empty observable
+        return of()
+      }),
+      // Combine all batches of transactions into a single array
+      reduce(
+        (allTransactions, batch) => allTransactions.concat(batch),
+        [] as TransactionLogEventWithEffects[],
+      ),
+    )
+    .pipe(
+      map((transactions) => removeDupes(transactions, cachedTransactions)),
+      tap((transactions) => {
+        documentTransactionsCache[cacheKey] = transactions
+      }),
+    )
+}
+
+interface EditEventsObservableValue {
+  editEvents: (EditReleaseEvent | CreateReleaseEvent)[]
+  loading: boolean
+  error: null | Error
+}
+export const INITIAL_VALUE: EditEventsObservableValue = {
+  editEvents: [],
+  loading: true,
+  error: null,
+}
+
+interface getReleaseActivityEventsOpts {
+  client: SanityClient
+  releaseId: string
+  releasesState$: Observable<ReleasesReducerState>
+}
+export function getReleaseEditEvents({
+  client,
+  releaseId,
+  releasesState$,
+}: getReleaseActivityEventsOpts): {
+  editEvents$: Observable<EditEventsObservableValue>
+} {
+  return {
+    editEvents$: releasesState$.pipe(
```

Comment:
do you envision more properties being added here? Should be fine to just return the observable (but also ok keep it this way)

Reply:
I debated myself on this one, I considered adding in a future an option to subscribe to the transactions received by the releases listener, instead of having to re fetch the transactions when the `_rev` changes (similar to what we have in the documents). 
But not sure how feasible that is and if it will ever happen. I will update it to only return the observable to make it simpler in the consumer.